cmd/nerdctl/image_convert.go

```diff
@@ -33,144 +33,164 @@ import (
 	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
 	"github.com/pkg/errors"
 	"github.com/sirupsen/logrus"
-	"github.com/urfave/cli/v2"
+	"github.com/spf13/cobra"
 )
 
-// imageConvertCommand is from https://github.com/containerd/stargz-snapshotter/blob/d58f43a8235e46da73fb94a1a35280cb4d607b2c/cmd/ctr-remote/commands/convert.go
-var imageConvertCommand = &cli.Command{
-	Name:      "convert",
-	Usage:     "convert an image",
-	ArgsUsage: "[flags] <source_ref> <target_ref>...",
-	Description: `Convert an image format.
+const imageConvertHelp = `Convert an image format.
 
 e.g., 'nerdctl image convert --estargz --oci example.com/foo:orig example.com/foo:esgz'
 
 Use '--platform' to define the output platform.
 When '--all-platforms' is given all images in a manifest list must be available.
-`,
-	Flags: []cli.Flag{
-		// estargz flags
-		&cli.BoolFlag{
-			Name:  "estargz",
-			Usage: "Convert legacy tar(.gz) layers to eStargz for lazy pulling. Should be used in conjunction with '--oci'",
-		},
-		&cli.StringFlag{
-			Name:  "estargz-record-in",
-			Usage: "Read 'ctr-remote optimize --record-out=<FILE>' record file (EXPERIMENTAL)",
-		},
-		&cli.IntFlag{
-			Name:  "estargz-compression-level",
-			Usage: "eStargz compression level",
-			Value: gzip.BestCompression,
-		},
-		&cli.IntFlag{
-			Name:  "estargz-chunk-size",
-			Usage: "eStargz chunk size",
-			Value: 0,
-		},
-		// generic flags
-		&cli.BoolFlag{
-			Name:  "uncompress",
-			Usage: "Convert tar.gz layers to uncompressed tar layers",
-		},
-		&cli.BoolFlag{
-			Name:  "oci",
-			Usage: "Convert Docker media types to OCI media types",
-		},
-		// platform flags
-		&cli.StringSliceFlag{
-			Name:  "platform",
-			Usage: "Convert content for a specific platform",
-			Value: &cli.StringSlice{},
-		},
-		&cli.BoolFlag{
-			Name:  "all-platforms",
-			Usage: "Convert content for all platforms",
-		},
-	},
-	Action: func(context *cli.Context) error {
-		var (
-			convertOpts = []converter.Opt{}
-		)
-		srcRawRef := context.Args().Get(0)
-		targetRawRef := context.Args().Get(1)
-		if srcRawRef == "" || targetRawRef == "" {
-			return errors.New("src and target image need to be specified")
-		}
+`
 
-		srcNamed, err := refdocker.ParseDockerRef(srcRawRef)
-		if err != nil {
-			return err
+// imageConvertCommand is from https://github.com/containerd/stargz-snapshotter/blob/d58f43a8235e46da73fb94a1a35280cb4d607b2c/cmd/ctr-remote/commands/convert.go
+func newImageConvertCommand() *cobra.Command {
+	imageConvertCommand := &cobra.Command{
+		Use:               "convert [flags] <source_ref> <target_ref>...",
+		Short:             "convert an image",
+		Long:              imageConvertHelp,
+		Args:              cobra.MinimumNArgs(2),
+		RunE:              imageConvertAction,
+		ValidArgsFunction: imageConvertShellComplete,
+	}
+
+	// #region estargz flags
```

Comment:
Is this for some IDE?

Reply:
It's supported by VSCode and IntelliJ IDEs.